Reduce use of strlen() in the codebase

cdumez opened the pull request:
#### 4dda9081d58ad072836beddaf27a8823ef52b7b1
<pre>
Reduce use of strlen() in the codebase
<a href="https://bugs.webkit.org/show_bug.cgi?id=286559">https://bugs.webkit.org/show_bug.cgi?id=286559</a>

Reviewed by Darin Adler.

* Source/WTF/wtf/DataLog.cpp:
(WTF::initializeLogFileOnce):
* Source/WTF/wtf/StringPrintStream.cpp:
(WTF::StringPrintStream::toCString const):
(WTF::StringPrintStream::tryToString const):
(WTF::StringPrintStream::toString const):
(WTF::StringPrintStream::toStringWithLatin1Fallback const):
* Source/WTF/wtf/cocoa/FileSystemCocoa.mm:
(WTF::FileSystemImpl::openTemporaryFile):
* Source/WTF/wtf/dtoa/utils.h:
(WTF::double_conversion::StringBuilder::Finalize):
* Source/WTF/wtf/text/StringCommon.h:
(WTF::requires):
* Source/WebCore/Modules/fetch/FetchBodyConsumer.cpp:
(WebCore::FetchBodyConsumer::packageFormData):
* Source/WebCore/PAL/pal/text/TextEncodingRegistry.cpp:
(PAL::WTF_REQUIRES_LOCK):
* Source/WebCore/platform/ios/LegacyTileCache.mm:
(WebCore::LegacyTileCache::drawLayer):
* Source/WebKit/Shared/Cocoa/SandboxExtensionCocoa.mm:
(WebKit::SandboxExtension::createHandleForTemporaryFile):
* Tools/DumpRenderTree/mac/DumpRenderTree.mm:
(libraryPathForDumpRenderTree):
(runTestingServerLoop):

Canonical link: <a href="https://commits.webkit.org/289462@main">https://commits.webkit.org/289462@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ad4a0bc95082a758ca79b69d1732c655983f76d6

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/86969 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/6478 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/41321 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/91827 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/37706 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/89018 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/6755 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/14546 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/67227 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/24994 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/89971 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/5159 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/78722 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/47549 "Found 1 new API test failure: /WPE/TestSSL:/webkit/WebKitWebView/ephemeral-tls-errors (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/4939 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/33094 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/36821 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/79753 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/75434 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/33973 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/93713 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/85741 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/14129 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/10274 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/76032 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/14333 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/74575 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/75229 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/18524 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/19557 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/17982 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/7008 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/14148 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/19439 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/108235 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/13893 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/26048 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/17335 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/15675 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WTF/wtf/text/StringCommon.h:
```diff
@@ -106,6 +106,15 @@ inline std::span<const char8_t> span(const std::u8string& string)
+    while (span[i] != '\0')
+        ++i;
+    return i;
+}
```
Not sure this needs the `inline` since it’s a template.

Also wondering if we want to stop when we hit the end of the `span` and just return `span.length()` instead of doing an `abort`. I guess to be consistent with other helpers, it’s OK to do it this way.

cdumez replied:
I thought about it but decided to crash (safely) in this case since:
- strlen() would have done bad things (reading out of bounds to find the null terminator).

I can be convinced to handle this without crashing though.

cdumez replied:
I have removed the inline.

darinadler commented on Source/WebCore/platform/ios/LegacyTileCache.mm:
```diff
@@ -568,16 +569,16 @@ static bool shouldRepaintInPieces(const CGRect& dirtyRect, CGSRegionObj dirtyReg
-        char text[16];
-        snprintf(text, sizeof(text), "%d", layer.paintCount);
+        std::array<char, 16> text;
+        snprintf(text.data(), text.size(), "%d", layer.paintCount);
```
Maybe we should use `CFStringCreateWithFormat` instead of involving an array of `char` at all.

cdumez replied:
I have updated the code to use CFStringCreateWithFormat()

cdumez committed:
Reduce use of strlen() in the codebase
https://bugs.webkit.org/show_bug.cgi?id=286559

Reviewed by Darin Adler.

* Source/WTF/wtf/DataLog.cpp:
(WTF::initializeLogFileOnce):
* Source/WTF/wtf/StringPrintStream.cpp:
(WTF::StringPrintStream::toCString const):
(WTF::StringPrintStream::tryToString const):
(WTF::StringPrintStream::toString const):
(WTF::StringPrintStream::toStringWithLatin1Fallback const):
* Source/WTF/wtf/cocoa/FileSystemCocoa.mm:
(WTF::FileSystemImpl::openTemporaryFile):
* Source/WTF/wtf/dtoa/utils.h:
(WTF::double_conversion::StringBuilder::Finalize):
* Source/WTF/wtf/text/StringCommon.h:
(WTF::requires):
* Source/WebCore/Modules/fetch/FetchBodyConsumer.cpp:
(WebCore::FetchBodyConsumer::packageFormData):
* Source/WebCore/PAL/pal/text/TextEncodingRegistry.cpp:
(PAL::WTF_REQUIRES_LOCK):
* Source/WebCore/platform/ios/LegacyTileCache.mm:
(WebCore::LegacyTileCache::drawLayer):
* Source/WebKit/Shared/Cocoa/SandboxExtensionCocoa.mm:
(WebKit::SandboxExtension::createHandleForTemporaryFile):
* Tools/DumpRenderTree/mac/DumpRenderTree.mm:
(libraryPathForDumpRenderTree):
(runTestingServerLoop):

Canonical link: https://commits.webkit.org/289462@main
